packages/settings-library/src/settings_library/r_clone.py

```diff
@@ -0,0 +1,36 @@
+from enum import Enum
+
+from pydantic import Field
+from .base import BaseCustomSettings
+from .s3 import S3Settings
+
+
+class S3Provider(str, Enum):
+    AWS = "AWS"
+    CEPH = "CEPH"
+    MINIO = "MINIO"
+
+
+class _RequiredS3Settings(S3Settings):
+    S3_ENDPOINT: str
+    S3_ACCESS_KEY: str
+    S3_SECRET_KEY: str
+    S3_BUCKET_NAME: str
+    S3_SECURE: bool
+
+
+class RCloneSettings(BaseCustomSettings):
+    R_CLONE_S3: _RequiredS3Settings = Field(auto_default_from_env=True)
+    R_CLONE_PROVIDER: S3Provider
+    R_CLONE_STORAGE_ENDPOINT: str = Field(
+        ..., description="endpoint where storage is present"
+    )
+
+    R_CLONE_AIOHTTP_CLIENT_TIMEOUT_TOTAL: float = 20
+    R_CLONE_AIOHTTP_CLIENT_TIMEOUT_SOCK_CONNECT: float = 5
+
+    @property
+    def storage_endpoint(self) -> str:
+        if not self.R_CLONE_STORAGE_ENDPOINT.startswith("http"):
+            return f"http://{self.R_CLONE_STORAGE_ENDPOINT}"
```

Comment:
This might also be `https`, right? Depending on `S3_SECURE`

Reply:
Yes, and what is the point of having a property when we already have ``R_CLONE_STORAGE_ENDPOINT`` and can use e.g. validators?

PLease, have a careful look at the implementation of postgres settings, specificaly the cached_properties
https://github.com/ITISFoundation/osparc-simcore/blob/b969b8d7a2f4e0cb4bba15a7738a53c6c5960f2a/packages/settings-library/src/settings_library/postgres.py#L53


The convention we follow is:
a) split in HOST and PORT
b) create ``cached_property`` to assemble different urls.

We should not break this convention